.setup/bin/recreate_submissions.py

```diff
@@ -0,0 +1,294 @@
+    autograding_hidden_non_extra_credit,autograding_hidden_extra_credit,
+    submission_time,autograding_complete
+- INSERT INTO electronic_gradeable_data (g_id,user_id,g_version,
+    submission_time,autograding_complete) VALUES (?,?,?,?,false)
+
+COURSE.electronic_gradeable_version
+- columns: g_id,user_id,team_id,active_version,anonymous_leaderboard
+- UPDATE electronic_gradeable_version SET active_version=?
+    WHERE g_id=? AND user_id=?
+'''
+
+import argparse
+import getpass
+import json
+import os
+import shutil
+import sqlalchemy as sql
+import sys
+import typing
+
+from submitty_utils import dateutils as submitty_dateutils
+
+# load configuration files
+with open('/usr/local/submitty/config/submitty.json','r') as conf:
+    SUBMITTY_CONFIG_JSON = json.load(conf)
+with open('/usr/local/submitty/config/database.json','r') as conf:
+    SUBMITTY_DBCONFIG_JSON = json.load(conf)
+
+ARG_INSTALL_DIR = SUBMITTY_CONFIG_JSON['submitty_install_dir']
+ARG_DATA_DIR = SUBMITTY_CONFIG_JSON['submitty_data_dir']
+#ARG_DB_HOST = 'localhost'
+ARG_DB_HOST = SUBMITTY_DBCONFIG_JSON['database_host']
+ARG_DB_PORT = SUBMITTY_DBCONFIG_JSON['database_port']
+ARG_DB_USER = SUBMITTY_DBCONFIG_JSON['database_user']
+ARG_DB_PASS = SUBMITTY_DBCONFIG_JSON['database_password']
+
+# variables specifying gradeable to submit to
+ARG_SEMESTER = ''
+ARG_COURSE = ''
+ARG_GRADEABLE = ''
+
+# place with files to submit
+ARG_SUBMISSIONS = ''
+
+# only use active versions?
+ARG_ACTIVE_ONLY = False
+
+# display extra output
+ARG_VERBOSE = False
+
+# some file names to look for
+UASF_NAME = 'user_assignment_settings.json'
+UAAF_NAME = '.user_assignment_access.json'
+UST_NAME = '.submit.timestamp'
+
+def submit(semester:str, course:str, gradeable:str, user:str, data: str,
+        func_add_submission: typing.Callable[[str,str,int,str],None],
+        func_update_active: typing.Callable[[str,str,int],None]):
+    '''
+    Create a new submission as a student
+    semester,course,gradeable = specify gradeable to make submission to
+    user = user to submit as
+    data = path to directory containing the files for submission
+    '''
+    gradeable_path = os.path.join(ARG_DATA_DIR,'courses',semester,course,
+        'submissions',gradeable)
+    # make sure we are submitting to a real gradeable
+    assert os.path.exists(gradeable_path),gradeable_path
+    user_path = os.path.join(gradeable_path,user)
+    if not os.path.exists(user_path):
+        os.makedirs(user_path)
+    # load data from the user_assignment_settings.json file if it exists
+    if os.path.exists(os.path.join(user_path,UASF_NAME)):
+        with open(os.path.join(user_path,UASF_NAME),'r') as uasf:
+            UAS = json.load(uasf)
+    else: # initialize empty
+        UAS = {
+            'active_version': 0,
+            'history': []
+        }
+    highest_version = 0 if len(UAS['history']) == 0 else max(obj['version']
+        for obj in UAS['history'])
```

Comment:
Given that this statement has to be broken into two lines anyway, it would be better to avoid an inline if-statement here.